simple-marketplace/hooks/useMetadatabyMetadataId.ts

```diff
@@ -0,0 +1,54 @@
+/*
+
+useMetadataByMetadataId Hook
+Description: This hook calls metadataByMetadataById method from @mintbase-js/data to get the information of the current NFT opened on the BuyModal.
+
+*/
+
+import { metadataByMetadataId } from '@mintbase-js/data';
+import { useQuery } from 'react-query';
+import { parseYactoToNear } from '../lib/numbers';
+import { SelectedNft, TokenListData } from '../types/types';
+
+const useMetadataByMetadataId = ({
+  metadataId,
+}: SelectedNft): Partial<TokenListData> => {
+  const {
+    isLoading,
+    data: metadata,
+  } = useQuery('metadataByMetadataId', () => metadataByMetadataId(metadataId), {
+    retry: false,
+    staleTime: Infinity,
+  });
+
+  const firstListing = metadata?.data?.listings[0];
+
+  if (!firstListing || firstListing === null) {
+    return {
+      amountAvailable: 0,
+      tokensTotal: 0,
+      price: 0,
+      tokenId: null,
+    };
+  }
+
+  const { price } = firstListing;
+
+  const prices = metadata?.data?.listings.map((elm) => ({
+    price: elm.price,
+    tokenId: elm.token.token_id,
+  }));
+
+  return {
```

Comment:
we can just use select property on the query and return the data

on the query you add select: `getMetadatabyMetadataId`


on the function you add all this logic you adding here

and on this hook you just return the regular `metadata`